src/App.jsx

```diff
@@ -1,97 +1,30 @@
-                />
-
-                <Route exact path={routes.NotFound} element={<ErrorPage />} />
-
-                <Route
-                    exact
-                    path={`${routes.Feed}/*`}
-                    element={
-                        <BaseScreen>
-                            <Routes>
-                                <Route
-                                    exact
-                                    index
-                                    element={
-                                        <ProtectedRoute>
-                                            <FeedScreenPage />
-                                        </ProtectedRoute>
-                                    }
-                                />
+    window.onerror = (event) => {
+        toast(event.toString(), {
+            type: 'error'
+        });
+    };
 
-                                <Route
-                                    exact
-                                    path={`${routes.Profile}/:id`}
-                                    element={
-                                        <ProtectedRoute>
-                                            <ProfilePage id={profileID} />
-                                        </ProtectedRoute>
-                                    }
-                                />
+    window.onunhandledrejection = (e) => {
+        throw new Error(e.reason.message);
+    };
 
-                                <Route
-                                    exact
-                                    path={routes.Chat}
-                                    element={
-                                        <ProtectedRoute>
-                                            <ChatPage />
-                                        </ProtectedRoute>
-                                    }
-                                />
-
-                                <Route
-                                    exact
-                                    path={routes.EditProfile}
-                                    element={
-                                        <ProtectedRoute>
-                                            <EditProfilePage />
-                                        </ProtectedRoute>
-                                    }
-                                />
-
-                                <Route
-                                    path={routes.Error}
-                                    element={
-                                        <Navigate
-                                            to={routes.NotFound}
-                                            replace
-                                        />
-                                    }
-                                />
-                            </Routes>
-                        </BaseScreen>
-                    }
-                />
-            </Routes>
-        </Router>
+    return (
+        <>
+            <ToastContainer theme='dark' autoClose={5000} />
```

Comment:
You are doing much more than you should, you handled the error state and added the react-toastify which wasn't any near of the task scope, always stick to the task description, "Setup firebase" doesn't mean, add error handling :P

Reply:
Yup sorry about that, do I have to remove it or something. I was in a trance of coding :P